cli/src/main/kotlin/commands/EvaluatorCommand.kt

```diff
@@ -205,27 +194,41 @@ class EvaluatorCommand : CliktCommand(name = "evaluate", help = "Evaluate rules
             }
         }
 
+        if (syntaxCheck) {
+            if (Evaluator().checkSyntax(script)) {
+                return
+            } else {
+                println("Syntax check failed.")
+                throw ProgramResult(2)
+            }
+        }
+
+        var ortResultInput = ortFile?.readValue<OrtResult>()
+
+        repositoryConfigurationFile?.let {
+            ortResultInput = ortResultInput?.replaceConfig(it.readValue())
+        }
+
+        packageCurationsFile?.let {
+            ortResultInput = ortResultInput?.replacePackageCurations(it.readValue())
+        }
+
+        val finalOrtResult = requireNotNull(ortResultInput) {
+            "The '--ort-file' option is required unless the '--syntax-check' option is used."
+        }
```

Comment:
After `requireNotNull` `ortResultInput` is smart-cast to `OrtResult`, so if you move this directly below the assignment you don't need the `?` in the other code and also don't need to assign `finalOrtResult`.

Reply:
Unfortunately, Kotlin smart casts do not work for clikt options.